client/src/communication.rs

```diff
@@ -1,80 +1,31 @@
-use std::{
-    time::Duration
-};
+mod udp;
+mod http;
 
-use reqwest::Client;
-use reqwest::header::CONTENT_TYPE;
 
 use crate::models::{Task, Auth, TaskResult, SystemInformation};
+use http::{
+    get_request,
+    post_request,
+};
+use crate::communication::udp::{
+    get_request_udp,
+    post_request_udp,
+};
 
 // Note that URL is set by environment variable during compilation
+const PROTOCOL: &'static str = env!("PROTOCOL");
```

Comment:
Would perhaps be nice to have some kind of check of whether this is specified as a valid protocol at client startup or compilation, such as something like the following (Python pseudocode):
```
if PROTOCOL not in [UDP_PROTOCOL, HTTPS_PROTOCOL]:
    raise ValueError('Invalid protocol specified!')
```
so that an else case isn't needed everywhere the PROTOCOL variable is used. Would be nice if it was possible to check at compile time to prevent compilation with invalid protocols.

Reply:
This is a good idea, so spent a couple of hours on this today and tried to add some compile time verification working. No success unfortunately.. We can add it as an issue and try to solve it later!